source/elements/oneMKL/source/domains/spblas/data_types/destroy_dense_matrix.rst

```diff
@@ -0,0 +1,55 @@
+.. SPDX-FileCopyrightText: 2024 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_sparse_destroy_dense_matrix:
+
+destroy_dense_matrix
+====================
+
+Destroys a ``dense_matrix_handle_t`` object.
+
+.. rubric:: Description and Assumptions
+
+The ``oneapi::mkl::sparse::destroy_dense_matrix`` function frees the resources
+allocated for the handle.
+
+If a buffer was provided, its reference count is decremented.
+
+If a USM pointer was provided, the data is not free'd.
```

Comment:
Given that a `sycl::buffer &` is passed in and that the library does not own the user arrays, the behavior same between USM and buffers: the `val` array is user-owned and is not freed. Only the `dense_matrix_handle_t` internals allocated in `create_dense_matrix()` API and owned by the implementing library are freed.

Reply:
I think we can keep this discussion in https://github.com/oneapi-src/oneAPI-spec/pull/522#discussion_r1521604846